Test fails on junit-jupiter-engine 5.4.2

## Expected Behavior

I'm using the 5.3.1 version of `junit-jupiter-engine` and I would like to bump it to 5.4.2.
The tests work fine on 5.3.1, but fails on 5.4.2.


## Actual Behavior

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
jun 11, 2019 9:47:07 AM org.junit.platform.launcher.core.DefaultLauncher handleThrowable
WARNING: TestEngine with ID 'junit-jupiter' failed to discover tests
java.lang.NoClassDefFoundError: org/junit/jupiter/api/MethodOrdererContext
	at org.junit.jupiter.engine.discovery.DiscoverySelectorResolver.createJavaElementsResolver(DiscoverySelectorResolver.java:86)
	at org.junit.jupiter.engine.discovery.DiscoverySelectorResolver.resolve(DiscoverySelectorResolver.java:57)
	at org.junit.jupiter.engine.discovery.DiscoverySelectorResolver.resolveSelectors(DiscoverySelectorResolver.java:50)
	at org.junit.jupiter.engine.JupiterTestEngine.discover(JupiterTestEngine.java:69)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverEngineRoot(DefaultLauncher.java:168)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverRoot(DefaultLauncher.java:155)
	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:120)
	at org.apache.maven.surefire.junitplatform.TestPlanScannerFilter.accept(TestPlanScannerFilter.java:56)
	at org.apache.maven.surefire.util.DefaultScanResult.applyFilter(DefaultScanResult.java:102)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.scanClasspath(JUnitPlatformProvider.java:143)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.ClassNotFoundException: org.junit.jupiter.api.MethodOrdererContext
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 15 more

jun 11, 2019 9:47:07 AM org.junit.platform.launcher.core.DefaultLauncher handleThrowable
WARNING: TestEngine with ID 'junit-jupiter' failed to discover tests
java.lang.NoClassDefFoundError: org/junit/jupiter/api/MethodOrdererContext
	at org.junit.jupiter.engine.discovery.DiscoverySelectorResolver.createJavaElementsResolver(DiscoverySelectorResolver.java:86)
	at org.junit.jupiter.engine.discovery.DiscoverySelectorResolver.resolve(DiscoverySelectorResolver.java:57)
	at org.junit.jupiter.engine.discovery.DiscoverySelectorResolver.resolveSelectors(DiscoverySelectorResolver.java:50)
	at org.junit.jupiter.engine.JupiterTestEngine.discover(JupiterTestEngine.java:69)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverEngineRoot(DefaultLauncher.java:168)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverRoot(DefaultLauncher.java:155)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:150)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.ClassNotFoundException: org.junit.jupiter.api.MethodOrdererContext
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 13 more

Results:

Tests run: 0, Failures: 0, Errors: 0, Skipped: 0

------------------------------------------------------------------------
BUILD FAILURE
------------------------------------------------------------------------
Total time: 8.221 s
Finished at: 2019-06-11T09:47:08-03:00
Final Memory: 8M/34M
```

The unique change I've made:

```diff
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter</artifactId>
    <version>5.4.2</version>
    <scope>test</scope>
</dependency>
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter-engine</artifactId>
-    <version>5.3.1</version>
+   <version>5.4.2</version>
    <scope>test</scope>
</dependency>
```


## Specifications

  - Version: JDK 12, JavaFX 12
  - Platform: Windows 10 x64

--- a/README.md
+++ b/README.md
@@ -55,6 +55,7 @@ dependencies {
 
 ```gradle
 dependencies {
+    testCompile 'org.junit.jupiter:junit-jupiter-api:5.5.1'
     testCompile "org.testfx:testfx-junit5:4.0.15-alpha"
 }
 ```
@@ -125,6 +126,12 @@ your project. TestFX supports JUnit 4, JUnit 5, and Spock.
 ### JUnit 5
 
 ```xml
+<dependency>
+    <groupId>org.junit.jupiter</groupId>
+    <artifactId>junit-jupiter-api</artifactId>
+    <version>5.5.1</version>
+    <scope>test</scope>
+</dependency>
 <dependency>
     <groupId>org.testfx</groupId>
     <artifactId>testfx-junit5</artifactId>

--- a/subprojects/testfx-junit5/testfx-junit5.gradle
+++ b/subprojects/testfx-junit5/testfx-junit5.gradle
@@ -28,11 +28,9 @@ test {
 dependencies {
     compile project(":testfx-core")
 
-    // Removes compile warnings about missing enums.
-    compile('org.apiguardian:apiguardian-api:1.0.0')
-
-    compile 'org.junit.jupiter:junit-jupiter-api:5.3.1'
-    testRuntime 'org.junit.jupiter:junit-jupiter-engine:5.3.1'
+    compileOnly 'org.junit.jupiter:junit-jupiter-api:5.5.1'
+    testCompileOnly 'org.junit.jupiter:junit-jupiter-api:5.5.1'
+    testRuntime 'org.junit.jupiter:junit-jupiter-engine:5.5.1'
 
     if (JavaVersion.current().isJava11Compatible()) {
         testCompile "org.testfx:openjfx-monocle:jdk-11+26"
@@ -79,9 +77,10 @@ compileTestJava {
         doFirst {
             options.compilerArgs = [
                     '--module-path', classpath.asPath,
-                    '--add-modules', 'org.junit.jupiter.api,javafx.controls',
+                    '--add-modules', 'org.junit.jupiter.api,javafx.controls,hamcrest.core',
                     '--add-reads', "$moduleName=org.junit.jupiter.api",
                     '--add-reads', "$moduleName=javafx.controls",
+                    '--add-reads', "$moduleName=hamcrest.core",
                     '--patch-module', "$moduleName=" + files(sourceSets.test.java.srcDirs).asPath,
             ]
             classpath = files()

